### PR TITLE
Delete item (use case)

### PR DESCRIPTION
**Pull request by lexazhs**
implemented delete_item (use case) and moved add_item (interface adapter) out of 'collection' package.

### CODE REVIEW

File: src/use_case/collection/delete_item/DeleteItemInteractor.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        this.ingredientFactory = ingredientFactory;
+    }
+
+//know when to have success view and fail view.
+    @Override
+    public void execute(DeleteItemInputData deleteItemInputData) {
+        List<String> itemList = deleteItemInputData.getItems();
+        List<String> deletedItemList = new ArrayList<>();
+        if(itemList.isEmpty()) {
+            deletePresenter.prepareFailView("No input. Try agian.");
```
Comment (WujiMikeChen) [inline src/use_case/collection/delete_item/DeleteItemInteractor.java:31]: typo here

File: src/use_case/collection/delete_item/DeleteItemInteractor.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    public DeleteItemInteractor(DeleteItemDataAccessInterface deleteItemDataAccessObject,
+                                DeleteItemOutputBoundary deletePresenter,
+                                IngredientFactory ingredientFactory) {
+        this.deleteItemDataAccessObject = deleteItemDataAccessObject;
+        this.deletePresenter = deletePresenter;
+        this.ingredientFactory = ingredientFactory;
+    }
+
+//know when to have success view and fail view.
```
Comment (WujiMikeChen) [inline src/use_case/collection/delete_item/DeleteItemInteractor.java:25]: add more detail about the condition for success and fail view

File: src/use_case/collection/delete_item/DeleteItemInteractor.java (hunk 1 above)
Comment (lexazhs) [reply src/use_case/collection/delete_item/DeleteItemInteractor.java]: fixed

File: src/use_case/collection/delete_item/DeleteItemInteractor.java (hunk 2 above)
Comment (lexazhs) [reply src/use_case/collection/delete_item/DeleteItemInteractor.java]: done

**Review by WujiMikeChen — APPROVED** [review]

**Pull request merged**